clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -400,12 +412,33 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
             int totalBytes = 0;
             int totalCount = 0;
             FetchResponse response = new FetchResponse(resp.responseBody());
+
+            // Look for OffsetOutOfRange error in fetchResponse
+            Map<TopicPartition, Long> offsetOutOfRangePartitions = new HashMap();
+            for (Map.Entry<TopicPartition, FetchResponse.PartitionData> entry : response.responseData().entrySet()) {
```

Comment:
It's definitely not a strong preference and the performance aspect doesn't really concern me either way. I am just trying to think through the full implications. One implication of discarding fetches is that it would be impossible to make progress unless the user handles the exception (either by propagating it or seeking to another offset). It's not clear to me if this is desirable or not. I also wonder whether some of these seek cases you mention are handled currently. Wouldn't it be possible even with the current patch to do a seek while a fetch is pending? In that case, the out of range error would apply to the previous offset and not the current one. That suggests to me that we ought to be checking the consumed position in SubscriptionState before throwing the exception.